vector/src/main/java/im/vector/app/features/home/room/detail/TimelineFragment.kt

```diff
@@ -1613,11 +1613,10 @@ class TimelineFragment @Inject constructor(
                 views.includeRoomToolbar.roomToolbarContentView.isClickable = roomSummary.membership == Membership.JOIN
                 views.includeRoomToolbar.roomToolbarTitleView.text = roomSummary.displayName
                 avatarRenderer.render(roomSummary.toMatrixItem(), views.includeRoomToolbar.roomToolbarAvatarImageView)
-                views.includeRoomToolbar.roomToolbarDecorationImageView.render(roomSummary.roomEncryptionTrustLevel)
-                views.includeRoomToolbar.roomToolbarPresenceImageView.render(
-                        roomSummary.isDirect && matrixConfiguration.presenceSyncEnabled,
-                        roomSummary.directUserPresence
-                )
+                val showPresence = roomSummary.isDirect && matrixConfiguration.presenceSyncEnabled
+                views.includeRoomToolbar.roomToolbarPresenceImageView.render(showPresence, roomSummary.directUserPresence)
+                val shieldView = if (showPresence) views.includeRoomToolbar.roomToolbarTitleShield else views.includeRoomToolbar.roomToolbarAvatarShield
```

Comment:
weird to have the shield at a different place depending if the presence is available or not...
We definitely need to iterate with design team, as you said

Reply:
agreed, iOS and web also handled these badges differently